src/main/java/org/jabref/gui/genfields/GenFieldsCustomizerDialogView.java

```diff
@@ -0,0 +1,72 @@
+package org.jabref.gui.genfields;
+
+import javax.inject.Inject;
+
+import javafx.fxml.FXML;
+import javafx.scene.control.ButtonType;
+import javafx.scene.control.TextArea;
+
+import org.jabref.gui.DialogService;
+import org.jabref.gui.help.HelpAction;
+import org.jabref.gui.util.BaseDialog;
+import org.jabref.gui.util.ControlHelper;
+import org.jabref.logic.help.HelpFile;
+import org.jabref.logic.l10n.Localization;
+import org.jabref.preferences.PreferencesService;
+
+import com.airhacks.afterburner.views.ViewLoader;
+
+public class GenFieldsCustomizerDialogView extends BaseDialog<Void> {
+
+    @FXML private ButtonType resetButton;
+    @FXML private ButtonType helpButton;
+    @FXML private ButtonType okButton;
+    @FXML private ButtonType cancelButton;
+    @FXML private TextArea fieldsTextArea;
+
+    @Inject private DialogService dialogService;
+    @Inject private PreferencesService preferences;
+    private GenFieldsCustomizerDialogViewModel viewModel;
+
+    public GenFieldsCustomizerDialogView() {
+        this.setTitle(Localization.lang("Set General Fields"));
+        this.setResizable(true);
+        this.getDialogPane().setPrefSize(300, 650);
+
+        ViewLoader.view(this)
+                  .load()
+                  .setAsDialogPane(this);
+
+        HelpAction helpCommand = new HelpAction(HelpFile.GENERAL_FIELDS);
+        //HelpAction is written with Swing, not JavaFX, so runCommand() cannot be used normally.  Here I am reaching into
+        //the command and running execute.  When HelpAction is converted to JavaFX,
+        //the following will need to be changed.
+        ControlHelper.setAction(helpButton, getDialogPane(), event -> helpCommand.getCommand().execute());
```

Comment:
You can use the ActionFactory to create the help button directly with the appropriate icon

```
       ActionFactory factory = new ActionFactory(preferencesService.getKeyBindingRepository());
      helpButton= factory.createIconButton(StandardActions.HELP, helpCommand.getCommand());
```

Reply:
I tried to implement it with the ActionFactory class, but createIconButton returns a Button while helpButton is a ButtonType, so I get a compiler error.

On that note, I checked the JavaFX dialogs, and the only one that seems to implement a help button is src/main/java/org/jabref/gui/bibtexkeypattern/BibtexKeyPatternPanel, which does not use buttonTypes at all, even for the other buttons. Do you have any other ideas?

Reply:
Button Types are only needed for Dialog buttons which close the dialog, e.g cancel, ok or yes /no button
All other buttons are normal buttons,e.g the help button or the browse buttons which open a file dialog

Reply:
Got it. Am I right that you are recommending I use the ActionFactory as you described, and just change the variable type to Button (instead of ButtonType) in the GenFieldsCustomizerDialogView class?